src/smart-components/user/add-user-to-group/add-user-to-group-wizard.js

```diff
@@ -0,0 +1,293 @@
+import messages from '../../../Messages';
+import pathnames from '../../../utilities/pathnames';
+import {
+  applyPaginationToUrl,
+  defaultAdminSettings,
+  defaultSettings,
+  isPaginationPresentInUrl,
+  syncDefaultPaginationWithUrl,
+} from '../../../helpers/shared/pagination';
+import { useIntl } from 'react-intl';
+import { TableToolbarViewOld } from '../../../presentational-components/shared/table-toolbar-view-old';
+
+import { useFieldApi } from '@data-driven-forms/react-form-renderer';
+import PermissionsContext from '../../../utilities/permissions-context';
+import { fetchGroups, fetchRoleForUser } from '../../../redux/actions/group-actions';
+import { useHistory } from 'react-router-dom';
+import { getUserGroups } from '../../../helpers/group/group-helper';
+
+const FORM_ID = 'add-user-to-group-form';
+
+const FormTemplate = (props) => <Pf4FormTemplate {...props} showFormControls={false} formWrapperProps={{ id: FORM_ID }} />;
+
+const GroupTable = (conf) => {
+  const { input, meta: formMeta, user, setLength, ...props } = useFieldApi(conf);
+  const textFilterRef = useRef(null);
+  const { orgAdmin, userAccessAdministrator } = useContext(PermissionsContext);
+  const isAdmin = orgAdmin || userAccessAdministrator;
+  const dispatch = useDispatch();
+  const intl = useIntl();
+  const history = useHistory();
+  const formOptions = useFormApi();
+
+  const columns = [
+    { title: intl.formatMessage(messages.name), key: 'name', transforms: [sortable] },
+    { title: intl.formatMessage(messages.roles) },
+    { title: intl.formatMessage(messages.members) },
+    { title: intl.formatMessage(messages.lastModified), key: 'modified', transforms: [sortable] },
+  ];
+
+  const { groups, meta, filters, isLoading } = useSelector(
+    ({ groupReducer: { groups, isLoading } }) => ({
+      groups: [...(groups?.data?.filter(({ platform_default, admin_default } = {}) => !(platform_default || admin_default)) || [])],
+      meta: groups?.pagination || groups?.meta,
+      filters: groups?.filters,
+      isLoading,
+    }),
+    shallowEqual
+  );
+  const [pagination, setPagination] = useState({ ...(orgAdmin ? defaultAdminSettings : defaultSettings), ...meta });
+  const [filterValue, setFilterValue] = useState(filters.name || '');
+  const [removeGroupsList, setRemoveGroupsList] = useState([]);
+  const [data, setData] = useState(undefined);
+
+  const setCheckedItems = (newSelection) => {
+    const rows = input.value || [];
+    const selection = newSelection(rows)
+      .filter(({ platform_default: isPlatformDefault, admin_default: isAdminDefault }) => !(isPlatformDefault || isAdminDefault))
+      .map(({ uuid, name }) => ({ uuid, label: name }));
+    setLength(selection.length);
+
+    input.onChange(selection);
+  };
+
+  const fetchData = (options) => dispatch(fetchGroups({ ...options, inModal: false }));
+
+  useEffect(() => {
+    const syncedPagination = syncDefaultPaginationWithUrl(history, pagination);
```

Comment:
We should also disable syncing the pagination with the URL, it probably does not make sense for modal (in addition, the URL sync implementation is not even prepared for use on the Users page and at the same time in its modal, it will cause problems)